source/dagger/gameplay/ping_pong/player_scores.cpp

```diff
@@ -49,7 +50,13 @@ void PlayerScoresSystem::Run()
 
     if (ballOnField == 0)
     {
-        // TODO: set speed to be random in both directions
-        CreatePingPongBall(s_TileSize, ColorRGBA(1, 1, 1, 1), { rand()%10 + 4,rand()%10 + 4,0 },   { 0,rand()%(s_FieldHeight / 2),0 });
+        //If the new ball is to be created when the slow-mode power up is active , it is created with the slowed down speed
```

Comment:
it's better to use new field inside ball, something like speed multiplier. And you will be setting it to 1 or 0.33f

Reply:
Added a field to the PingPongBall component , so it isn't necessary to change ball speed every time